src/idl_gen_general.cpp

```diff
@@ -203,8 +203,13 @@ class GeneralGenerator : public BaseGenerator {
       const std::string &classcode, bool needs_includes) {
     if (!classcode.length()) return true;
 
-    std::string code;
-    code = code + "// " + FlatBuffersGeneratedWarning();
+    std::string code("//------------------------------------------------------------------------------\n"
```

Comment:
Pretty sure it wouldn't fail to recognize it as generated code because of missing `----`. See https://github.com/google/flatbuffers/issues/4287
Besides that, I had 2 other comments you could address:
You can still call FlatBuffersGeneratedWarning(); on the next line.
Also, I believe #4287 showed that there should be a </auto-generated> at the end of the file?